File: Lib/EntitySchema.cs
using System;
using System.Linq;

namespace System.Data.EntitySchema {

  public class EntitySchema {

    public string Name { get; set; } = "";
    public string NamePlural { get; set; } = "";
    public string InheritedEntityName { get; set; } = null;
    public string Summary { get; set; } = "";

    public bool IsBlEntrypoint { get; set; } = false;

    public string PrimaryKeyIndexName { get; set; } = "";
    public IndexSchema[] Indices { get; set; } = { };

    public FieldSchema[] Fields { get; set; } = { };

  }

}

File: Lib_JS/src/entitySchema.ts
import {IndexSchema} from "./indexSchema"
import {FieldSchema} from "./fieldSchema"

export class EntitySchema {
  
  public name: string = '';
  public namePlural: string = '';
  public inheritedEntityName: string|null = null;
  public summary: string = '';

  public isBlEntrypoint: boolean = false;

  public PrimaryKeyIndexName: string = '';
  public indices: IndexSchema[] = [];

  public fields: FieldSchema[] = [];

}
